pkg/reconciler/v1alpha1/revision/revision_test.go

```diff
@@ -740,38 +836,28 @@ func TestGlobalResyncOnConfigMapUpdate(t *testing.T) {
 
 			rev := getTestRevision()
 			revClient := servingClient.ServingV1alpha1().Revisions(rev.Namespace)
-			deploymentsClient := kubeClient.Apps().Deployments(rev.Namespace)
 			h := NewHooks()
 
-			h.OnUpdate(&servingClient.Fake, "revisions", func(obj runtime.Object) HookResult {
-				updatedRev := obj.(*v1alpha1.Revision)
-				t.Logf("Revision updated: %v", updatedRev.Name)
-				updatedDeployment, err := deploymentsClient.Get(resourcenames.Deployment(updatedRev), metav1.GetOptions{})
-				if err != nil {
-					t.Error(err)
-				}
-
-				got, wasUpdated := test.wasUpdated(test.expected, updatedRev, updatedDeployment)
-
-				if !wasUpdated {
-					t.Logf("No update occurred; expected: %s got: %s", test.expected, got)
-					return HookIncomplete
+			h.OnCreate(&kubeClient.Fake, "deployments", func(obj runtime.Object) HookResult {
+				result := test.callback(t)(obj)
+				if result == HookIncomplete {
+					t.Log("Registering OnUpdate hook")
+					h.OnUpdate(&kubeClient.Fake, "deployments", test.callback(t))
 				}
 				return HookComplete
```

Comment:
This could use a comment.  "We didn't see the state we wanted on Create, so look for it on Update".